### PR TITLE
Avoid loss for too short input to be infinity

### LINKED ISSUE #5614: CTC loss function for too long label sequences
The current implementation of CTC loss function returns positive infinity when it gets a long label sequence. In such case, the CTC function cannot find any paths from BOS to EOS, and the marginal probability of a given sentence becomes zero. So the loss defined as negative log likelihood is positive infinity.
To avoid such situation, for example Baidu's warp-ctc returns zero: https://github.com/baidu-research/warp-ctc/blob/master/include/detail/cpu_ctc.h#L488

### PR DESCRIPTION
**Pull request by unnonouno**
fix #5614

### CODE REVIEW

File: tests/chainer_tests/functions_tests/loss_tests/test_ctc.py [hunk 1]
```diff
@@ -250,4 +250,49 @@ def test_not_iterable(self):
...
+            xs, ts, 2, label_length=label_length,
+            reduce='no').data
+        assert loss[0] == 0
+        assert loss[1] > 0
+
+    def test_forward_cpu(self):
+        self.check_forward(
+            tuple(self.x), self.t, self.label_length)
+
+    def test_forward_gpu(self):
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/loss_tests/test_ctc.py]:
```suggestion
    @attr.gpu
    def test_forward_gpu(self):
```

File: tests/chainer_tests/functions_tests/loss_tests/test_ctc.py [hunk 2]
```diff
@@ -250,4 +250,49 @@ def test_not_iterable(self):
...
+
+        gradient_check.check_backward(
+            f, (label_length, ts) + xs, gy,
+            eps=1e-2, atol=1e-4)
+
+    def test_backward_cpu(self):
+        self.check_backward(
+            tuple(self.x), self.t, self.label_length, self.gy)
+
+    def test_backward_gpu(self):
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/loss_tests/test_ctc.py]:
```suggestion
    @attr.gpu
    def test_backward_gpu(self):
```

**Comment by toslunar** [comment]
ping @unnonouno

**Comment by unnonouno** [comment]
Thank you for reviewing. I heard that cuDNN's CTC also supports this case but its behavior is not defined in the document. I want to make behavior of my fix same as cuDNN's one. I'll check it after #5650 is merged.

**Pull request closed without merging**